src/assets/wise5/components/dialogGuidance/DialogGuidanceFeedbackRuleEvaluator.ts

```diff
@@ -117,11 +117,30 @@ export class DialogGuidanceFeedbackRuleEvaluator {
   private evaluateTerm(term: string, response: CRaterResponse): boolean {
     if (this.isHasKIScoreTerm(term)) {
       return this.evaluateHasKIScoreTerm(term, response);
+    } else if (this.isIdeaCountTerm(term)) {
+      return this.evaluateIdeaCountTerm(term, response);
     } else {
       return this.evaluateIdeaTerm(term, response);
     }
   }
 
+  private isIdeaCountTerm(term: string): boolean {
+    return /ideaCount(MoreThan|Equals|LessThan)\([\d+]\)/.test(term);
+  }
+
+  private evaluateIdeaCountTerm(term: string, response: CRaterResponse): boolean {
+    const matches = term.match(/ideaCount(.*)\((.*)\)/);
+    const comparer = matches[1];
+    const expectedIdeaCount = parseInt(matches[2]);
+    if (comparer === 'MoreThan') {
+      return response.getDetectedIdeaCount() > expectedIdeaCount;
+    } else if (comparer === 'Equals') {
+      return response.getDetectedIdeaCount() === expectedIdeaCount;
+    } else {
+      return response.getDetectedIdeaCount() < expectedIdeaCount;
+    }
```

Comment:
Should we use a switch here?

```
    switch (comparer) {
      case 'MoreThan':
        return response.getDetectedIdeaCount() > expectedIdeaCount;
      case 'Equals':
        return response.getDetectedIdeaCount() === expectedIdeaCount;
      case 'LessThan':
        return response.getDetectedIdeaCount() < expectedIdeaCount;
      default:
        return false;
    }
```

Reply:
Hmm. This increases the complexity of this function from 8 to 10 because of the "default" case. 

How about 
```
   switch (comparer) {
      case 'MoreThan':
        return response.getDetectedIdeaCount() > expectedIdeaCount;
      case 'Equals':
        return response.getDetectedIdeaCount() === expectedIdeaCount;
      default:
        return response.getDetectedIdeaCount() < expectedIdeaCount;
    }
```
since we can assume that those are the only possible values that comparer could be inside this function?

Reply:
I think it would be better to be more explicit and have a 'LessThan' case as well as a default.

Reply:
We would never reach the default case in your code example because the values for comparer can only be "MoreThan", "Equals", or "LessThan". ```evaluateIdeaCountTerm()``` is a private function, and it is only called when ```isIdeaCountTerm()``` is true, which guarantees that the comparer is one of the three possible values.

If you want to add a default case, we can throw an error instead of returning a false value. Returning a false value will allow the program to continue, which may result in unwanted/unexpected behaviors down the line.

```
    switch (comparer) {
      case 'MoreThan':
        return response.getDetectedIdeaCount() > expectedIdeaCount;
      case 'Equals':
        return response.getDetectedIdeaCount() === expectedIdeaCount;
      case 'LessThan':
        return response.getDetectedIdeaCount() < expectedIdeaCount;
      default:
        throw new Error("Unexpected comparer value.");
    }
```

Reply:
Yea, throwing an error there is good.